src/main/scala/com/cognite/sdk/scala/v1/resources/files.scala

```diff
@@ -158,33 +147,21 @@ class Files[F[_]: Applicative](val requestSession: RequestSession[F])
         )
       )
 
-  def download(item: FileDownload, out: java.io.OutputStream): F[Unit] = {
-    val link: F[FileDownloadLink] = downloadLink(item)
-
-    requestSession.flatMap(
-      link,
-      (link: FileDownloadLink) => {
-        val response = requestSession.send { request =>
-          request
-            .get(
-              uri"${link.downloadUrl}"
-            )
-            .response(asByteArray)
-        }
-        requestSession.map(
-          response,
-          (res: Response[Either[String, Array[Byte]]]) =>
-            res.body match {
-              case Right(bytes) => out.write(bytes)
-              case Left(_) =>
-                throw SdkException(
-                  s"File download of file ${item.toString} failed with error code ${res.code.toString}"
-                )
-            }
-        )
+  def download(item: FileDownload, out: java.io.OutputStream): F[Unit] =
+    for {
+      link <- downloadLink(item)
+      res <- requestSession.send { request =>
+        request
+          .get(uri"${link.downloadUrl}")
+          .response(asByteArray)
       }
-    )
-  }
+      bytes <- F.fromOption(
+        res.body.toOption,
+        SdkException(
+          s"File download of file ${item.toString} failed with error code ${res.code.toString}"
+        )
+      )
+    } yield out.write(bytes)
```

Comment:
I think we should also wrap `.write` into `F.blocking {}` as we don't know if outputstream would block or take long time

Reply:
good idea!
but I think `Blocking` comes from `Sync`.... which would mean `IO` rather than `Either` (I guess we could add the Sync requirement for that method only)